dask_cuda/disk_io.py

```diff
@@ -200,24 +204,31 @@ def disk_read(header: Mapping, gds=False) -> list:
     frames: list
         List of read frames
     """
-    ret = []
+    ret: list = []
     if gds:
         import kvikio  # isort:skip
 
+        ret.extend(
+            get_new_cuda_buffer()(length)
+            if is_cuda
+            else np.empty((length,), dtype="u1")
+            for length, is_cuda in zip(header["frame-lengths"], header["cuda-frames"])
+        )
         with kvikio.CuFile(header["path"], "rb") as f:
+            futs = []
             file_offset = 0
-            for length, is_cuda in zip(header["frame-lengths"], header["cuda-frames"]):
-                if is_cuda:
-                    buf = get_new_cuda_buffer()(length)
-                else:
-                    buf = np.empty((length,), dtype="u1")
-                f.pread(
-                    buf=buf, count=length, file_offset=file_offset, buf_offset=0
-                ).get()
-                file_offset += length
-                ret.append(buf)
+            for b in ret:
+                n = b.nbytes
+                futs.append(
+                    f.pread(buf=b, count=n, file_offset=file_offset, buf_offset=0)
+                )
+                file_offset += n
```

Comment:
This could be replaced by a single vectorized read once available ( https://github.com/rapidsai/kvikio/issues/77 )